clusterman/batch/autoscaler.py

```diff
@@ -135,6 +137,10 @@ def _autoscale(self):
     def run(self):
         while self.running:
             try:
+                logger.info('Reloading watcher config files')
+                if self.reload_watchers():
+                    self.initialize_autoscaler()
```

Comment:
I'm wondering how expensive this is. There might be some things in here that we don't want to do again, like recreating metrics things. It might more worth it to create a function in the Autoscaler reloads everything that is dependent on srv-configs. Or better yet, instead of having the Autoscaler save a lot of srv-configs values into local variables during its init functions, just turn each of those local-variable uses into a srv-configs read (this is how it normally works).

Ditto for the PoolManagers.

Reply:
Most of the Autoscaler constructor actually reads from srv-configs. For example, both `ClustermanMetricsBotoClient` and `Signal` are objects that are instantiated in the Autoscaler connector. Initializing `ClustermanMetricsBotoClient` reads a `ttl_days` field and initializing Signal reads various metrics fields. I think it would it would be quite messy to just pick and choose properties to update when a lot of those properties are either inherited from `Autoscaler` or `ClusterMetricsCollector` or in an external package. In addition, even if it is an expensive operation, we aren't executing it unless something in srv-config actually changes which isn't often. Anyways, we could use more opinions on this.

Reply:
The way this used to work is that if a srv-configs file changed, yelp_batch would just set `self.running` to false and the entire batch would restart.  (see https://docs.google.com/forms/d/1McU7891ROmbh2ugtHE-9oczNDvgWyCTF59NOhZtqJUg/edit).  Is this no longer how yelp_batch works?  I haven't been following the changes too closely.  In any case, I think I would prefer that solution still to reloading srv-configs piecemeal.  There's a lot of saved state in Clusterman, and I worry about forgetting to update some of it.  Plus, it makes this code simpler.

Reply:
@drmorr0 You might have linked something else there. Also, how does this make the code simpler? Looking at the constructor of both PoolManager and Autoscaler, it seems like most of its attributes require reading from staticconf anyways (https://github.com/Yelp/clusterman/blob/master/clusterman/autoscaler/autoscaler.py#L84-L110). Also there are multiple child classes that inherit from these classes that would make re-initializing kinda messy. It's fair that there is a lot of saved state in clusterman and reloading the autoscaler might forget some of it. My point is that bulk of initializing the class `Autoscaler` requires reading from staticconf anyays, so why bother with pick and choosing?